File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        maxProfit = 0

        for right, num in enumerate(prices):
            leftPrice = prices[left]
            rightPrice = prices[right]

            currProfit = rightPrice - leftPrice
            maxProfit = max(currProfit, maxProfit)

            if leftPrice > rightPrice:
                left = right

        return maxProfit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let left = 0;
    let maxProfit = 0;

    for (let right = 0; right < prices.length; right++) {
        const leftPrice = prices[left];
        const rightPrice = prices[right];
        const currProfit = rightPrice - leftPrice;
        maxProfit = Math.max(currProfit, maxProfit);

        if (leftPrice > rightPrice) {
            left = right
        }
    };

    return maxProfit;
};